core/src/main/scala/akka/persistence/r2dbc/state/scaladsl/R2dbcDurableStateStore.scala

```diff
@@ -125,7 +151,41 @@ class R2dbcDurableStateStore[A](system: ExtendedActorSystem, config: Config, cfg
       manifest,
       if (tag.isEmpty) Set.empty else Set(tag))
 
-    stateDao.upsertState(serializedRow, value)
+    val serializedChangedEvent: Option[SerializedJournalRow] = {
+      changeEvent.map { event =>
+        val eventAnyRef = event.asInstanceOf[AnyRef]
+        val serializedEvent = eventAnyRef match {
+          case s: SerializedEvent => s // already serialized
+          case _ =>
+            val bytes = serialization.serialize(eventAnyRef).get
+            val serializer = serialization.findSerializerFor(eventAnyRef)
+            val manifest = Serializers.manifestFor(serializer, eventAnyRef)
+            new SerializedEvent(bytes, serializer.identifier, manifest)
+        }
+
+        val entityType = PersistenceId.extractEntityType(persistenceId)
+        val slice = persistenceExt.sliceForPersistenceId(persistenceId)
+        val timestamp = if (journalSettings.useAppTimestamp) InstantFactory.now() else JournalDao.EmptyDbTimestamp
+
+        SerializedJournalRow(
+          slice,
+          entityType,
+          persistenceId,
+          revision,
```

Comment:
Can't we just disallow that?

Reply:
Do you mean that we would read latest sequence number from journal and thereby check that there is no gap? That has much overhead and adding caching and such is not trivial (as in the EventWriter).

I'm thinking that we might need something to allow starting from later offset without requiring previous for other reasons. E.g. came up in a question about migration from jdbc, or as soon as you have started to delete events and later add a new projection. I'm not sure if it would be some kind of relaxed offset validation or some kind of offset population job that can be run as a migration task.